src/overlays/actors/ovl_Obj_Mine/z_obj_mine.c

```diff
@@ -24,131 +53,1182 @@ ActorInit Obj_Mine_InitVars = {
+    Vec3f nextWaypoint;
+    f32 distToWaypoint;
+    f32 step;
+    f32 target;
+    s32 bgId; // not used
+
+    // thisx->velocity is temporarily set to the vector difference to the next waypoint.
+    Math_Vec3s_ToVec3f(&nextWaypoint, &this->waypoints[this->waypointIndex + 1]);
+    Math_Vec3f_Diff(&nextWaypoint, &thisx->world.pos, &thisx->velocity);
+    distToWaypoint = Math3D_Vec3fMagnitude(&thisx->velocity);
+
+    // Mine slows to 2.0f speed when 8 frames away from the next waypoint
+    if ((distToWaypoint < (this->pathSpeed * 8.0f)) && (this->pathSpeed > 2.0f)) {
+        target = 2.0f + ((this->pathSpeed - 2.0f) * 0.1f);
+        step = this->pathSpeed * 0.03f;
+    } else {
+        target = this->pathSpeed;
+        step = this->pathSpeed * 0.16f;
+    }
+    Math_StepToF(&this->actor.speed, target, step);
+
+    // Checks if mine will reach the waypoint next frame
+    if ((this->actor.speed + 0.05f) < distToWaypoint) {
+        // Rescales thisx->velocity to be equal in magnitude to speed
+        Math_Vec3f_Scale(&thisx->velocity, this->actor.speed / distToWaypoint);
+        this->actor.world.pos.x += thisx->velocity.x;
+        this->actor.world.pos.y += thisx->velocity.y;
+        this->actor.world.pos.z += thisx->velocity.z;
+    } else {
+        this->actor.speed *= 0.4f;
+        this->waypointIndex++;
+        if (this->waypointIndex >= this->waypointCount) {
+            this->waypointIndex = 0;
+        }
+        ObjMine_Path_MoveToWaypoint(this, this->waypointIndex);
+    }
+    this->actor.floorHeight =
+        BgCheck_EntityRaycastFloor5(&play->colCtx, &this->actor.floorPoly, &bgId, &this->actor, &thisx->world.pos);
+    if (this->actor.flags & ACTOR_FLAG_40) {
+        Vec3f rotAxis;
+        Vec3f yhatCrossV;
+        MtxF rotMtxF;
+
+        // Makes mines appear to roll while traversing the path
+        Math3D_CrossProduct(&sStandardBasis.y, &thisx->velocity, &yhatCrossV);
+        if (ObjMine_GetUnitVec3f(&yhatCrossV, &rotAxis)) {
+            Matrix_RotateAxisF(this->actor.speed / PATH_RADIUS, &rotAxis, MTXMODE_NEW);
+            Matrix_RotateYS(this->actor.shape.rot.y, MTXMODE_APPLY);
+            Matrix_RotateXS(this->actor.shape.rot.x, MTXMODE_APPLY);
+            Matrix_RotateZS(this->actor.shape.rot.z, MTXMODE_APPLY);
+            Matrix_Get(&rotMtxF);
+            Matrix_MtxFToYXZRot(&rotMtxF, &this->actor.shape.rot, false);
+        }
+    }
+}
+
+void ObjMine_SetupExplode(ObjMine* this) {
+    this->actor.flags |= ACTOR_FLAG_10;
+    this->actor.draw = ObjMine_DrawExplosion;
+    this->actor.shape.shadowDraw = NULL;
+    this->actor.scale.x = 0.02f;
+    this->actor.scale.y = 0.02f;
+    this->actor.scale.z = 0.02f;
+    this->actionFunc = ObjMine_Explode;
+}
+
+void ObjMine_Explode(ObjMine* this, PlayState* play) {
+    this->actor.scale.x *= 1.8f;
+    if (this->actor.scale.x > 0.02f * 1.5f * 5.832f) { // 5.832 = 1.8^3
```

Comment:
Hmm, curious calculation, probably this originally had defines. Does this mean it expands for 3 frames then?